src/pages/productList/ProductList.jsx

```diff
@@ -1,20 +1,22 @@
 import prodList from "./productList.module.css";
 import filter from "./filter.module.css";
 import foot from "../../components/footer/footer.module.css";
-import { categories } from "../home/Home";
-
+import { Error } from "../../components";
 import { useState, useEffect } from "react";
 import axios from "axios";
 
 export function ProductList() {
   const [productList, setProductList] = useState([]);
+  const [categories, setCategories] = useState([]);
   const [error, setError] = useState(false);
 
   useEffect(() => {
     (async function () {
       try {
         let products = await axios.get("/api/products");
         setProductList(products.data.products);
+        let categories = await axios.get("/api/categories");
+        setCategories(categories.data.categories);
```

Comment:
try using it with than it will have less chances of throwing error. Suppose categories  is null than categories.data will give you throw an exceptionit would be like null.data